lib/testing/integration/authservice.go

```diff
@@ -270,6 +290,35 @@ func (auth *AuthService) IsReady() bool {
 	return auth.isReady
 }
 
+// setCAPin sets the auth service CA Pin using output from tctl.
+func (auth *AuthService) setCAPin(ctx context.Context) error {
+	log := logger.Get(ctx)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, auth.tctlPath, "status", "--config", auth.configPath, "--auth-server", auth.AuthAddr().String())
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		// Split tctl output lines and print line by line.
+		splitOutput := strings.Split(string(output), "\\n")
+		for _, line := range splitOutput {
+			log.Debug("AuthService.setCAPin log: ", line)
+		}
+		return trace.Wrap(err, "failed to get auth status")
+	}
+
+	submatch := regexpAuthCAPin.FindStringSubmatch(string(output))
+	if submatch != nil {
+		auth.mu.Lock()
+		defer auth.mu.Unlock()
+		auth.caPin = submatch[1]
+	}
+	if auth.caPin == "" {
+		return trace.Errorf("failed to find CA Pin in auth status")
+	}
```

Comment:
Can this be a method in `tctl` package?

Reply:
Good idea, I moved all of the logic to `integration` and `setup` instead of `authservice`.